pkg/client/client_test.go

```diff
@@ -1178,6 +1178,34 @@ var _ = Describe("Client", func() {
 				close(done)
 			})
 
+			It("should patch correctly after a status patch", func(done Done) {
```

Comment:
err... what does "correctly" mean here?  I feel like I'm missing some context for this.

Reply:
Apologies, we've noticed that the other way around, doing Status.Patch after Patch makes the object lose status information. I added this test to ensure that, going forward, this behavior is correct.

I can also add the opposite (failing) test and mark it to be Skipped if you prefer

Reply:
ah, I think you should just elaborate a bit in the test description

Reply:
Sounds good, I reworded a little yesterday but can add more comments.

FYI, we ended up adding a package to wrap the Patch logic for us https://github.com/kubernetes-sigs/cluster-api/pull/1275